cub/test/CMakeLists.txt

```diff
@@ -276,6 +279,9 @@ function(cub_add_test target_name_var test_name test_src cub_target launcher_id)
 
     add_executable(${test_target} "${test_src}")
     target_link_libraries(${test_target} ${cub_target})
+    if (${test_src} MATCHES "test_nvtx_in_usercode.cu")
+      target_link_libraries(${test_target} nvtx3-cpp)
+    endif()
```

Comment:
How can I best add a library dependency to only a single test?

Reply:
I do something like this: https://github.com/NVIDIA/cccl/blob/ca7109ab34eea98b37f1cd9359aa327fb37ccae2/cub/test/CMakeLists.txt#L252-L257 

@alliepiper might have a better idea

Reply:
This is indeed the best way for now. Thrust has a fancier mechanism that lets us add a test-specific `${test_name}.cmake` file that gets loaded and applied to the test targets, but that doesn't currently exist in CUB.